ddl/index.go

```diff
@@ -626,30 +632,35 @@ func onDropIndex(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 				adjustColumnInfoInDropColumn(tblInfo, firstHiddenOffset)
```

Comment:
Is it still right even if the dependent columns are duplicate? That is, when multiple indexes depend on the same hidden column.

Reply:
I'm sure that different expression indexes don't depend on the same hidden column. @djshow832

Reply:
The expression's index dependent hidden columns are continuous. Since there are many indexes now, we can't just reuse the previous logic.

Reply:
> The expression's index dependent hidden columns are continuous. Since there are many indexes now, we can't just reuse the previous logic.

I refer to the code of DropColumns and rewrite the logic of this.